src/platform/Linux/NetworkCommissioningDriver.cpp

```diff
@@ -0,0 +1,269 @@
+ */
+
+#include <lib/support/CodeUtils.h>
+#include <lib/support/SafeInt.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/Linux/NetworkCommissioningDriver.h>
+#include <platform/Linux/ThreadStackManagerImpl.h>
+#include <platform/ThreadStackManager.h>
+
+#include <string>
+#include <vector>
+
+using namespace chip;
+using namespace chip::Thread;
+
+namespace chip {
+namespace DeviceLayer {
+namespace NetworkCommissioning {
+
+#if CHIP_DEVICE_CONFIG_ENABLE_WPA
+CHIP_ERROR LinuxWiFiDriver::CommitConfiguration()
+{
+    mRunningNetwork = mStagingNetwork;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR LinuxWiFiDriver::RevertConfiguration()
+{
+    mStagingNetwork = mRunningNetwork;
+    return CHIP_NO_ERROR;
+}
+
+Status LinuxWiFiDriver::AddOrUpdateNetwork(ByteSpan ssid, ByteSpan credentials)
+{
+    VerifyOrReturnError(
+        mStagingNetwork.ssidLen == 0 ||
+            (ssid.size() == mStagingNetwork.ssidLen && memcmp(ssid.data(), mStagingNetwork.ssid, mStagingNetwork.ssidLen) == 0),
+        Status::kBoundsExceeded);
+
+    VerifyOrReturnError(credentials.size() <= sizeof(mStagingNetwork.credentials), Status::kOutOfRange);
+    memcpy(mStagingNetwork.credentials, credentials.data(), credentials.size());
+
+    using WiFiCredentialsLenType = decltype(mStagingNetwork.credentialsLen);
+    VerifyOrReturnError(CanCastTo<WiFiCredentialsLenType>(credentials.size()), Status::kOutOfRange);
+    mStagingNetwork.credentialsLen = static_cast<WiFiCredentialsLenType>(credentials.size());
+
+    VerifyOrReturnError(ssid.size() <= sizeof(mStagingNetwork.ssid), Status::kOutOfRange);
+    memcpy(mStagingNetwork.ssid, ssid.data(), ssid.size());
+
+    using WiFiSSIDLenType = decltype(mStagingNetwork.ssidLen);
+    VerifyOrReturnError(CanCastTo<WiFiSSIDLenType>(ssid.size()), Status::kOutOfRange);
+    mStagingNetwork.ssidLen = static_cast<WiFiSSIDLenType>(ssid.size());
+
+    return Status::kSuccess;
+}
+
+Status LinuxWiFiDriver::RemoveNetwork(ByteSpan networkId)
+{
+    VerifyOrReturnError(networkId.size() == mStagingNetwork.ssidLen &&
+                            memcmp(networkId.data(), mStagingNetwork.ssid, mStagingNetwork.ssidLen) == 0,
+                        Status::kNetworkNotFound);
+
+    mStagingNetwork.ssidLen = 0;
+    return Status::kSuccess;
+}
+
+Status LinuxWiFiDriver::ReorderNetwork(ByteSpan networkId, uint8_t index)
+{
+    // We only support one network, so reorder is actually no-op.
+    VerifyOrReturnError(networkId.size() == mStagingNetwork.ssidLen &&
+                            memcmp(networkId.data(), mStagingNetwork.ssid, mStagingNetwork.ssidLen) == 0,
+                        Status::kNetworkNotFound);
+
+    mStagingNetwork.ssidLen = 0;
+    return Status::kSuccess;
+}
+
+void LinuxWiFiDriver::ConnectNetwork(ByteSpan networkId, ConnectCallback * callback)
+{
+    CHIP_ERROR err          = CHIP_NO_ERROR;
+    Status networkingStatus = Status::kSuccess;
+
+    // We only support one network, so reorder is actually no-op.
```

Comment:
Cut'n'paste mistake on this comment.

Also, it is legal to ConnectNetwork without the staging (i.e., outside of an armed fail-safe). Seems like the check here prevents that.

Reply:
Added a comment in the beginning of the file for the mSavedNetwork & mStagingNetwork -- mStagingNetwork always has a value and is set to mSavedNetwork when starting up. also it will be written back to mSavedNetwork on commit and set to mSavedNetwork on revert.